src/JsonKnownTypes/JsonKnownTypesSettingsManager.cs

```diff
@@ -23,8 +30,9 @@ internal static DiscriminatorValues GetDiscriminatorValues<T>()
         {
             var discriminatorAttribute = AttributesManager.GetJsonDiscriminatorAttribute(typeof(T));
 
-            var discriminatorSettings = discriminatorAttribute == null 
-                ? DefaultDiscriminatorSettings 
+            var discriminatorSettings = discriminatorAttribute == null
+                ? JsonKnownTypesSettingsManager<T>.DefaultDiscriminatorSettings
```

Comment:
This will be null unless you explicitly set generic settings. For example: `JsonKnownTypesSettingsManager<ClassWithDuplicateChild>.DefaultDiscriminatorSettings.NameDiscriminatorResolver = type => type.FullName;`

So, first it looks for the attribute, if that is null, it will look for a specific setting for the generic type (`<ClassWithDuplicateChild>` in the example), if that is also null it will fall back to the default settings.

At least thats how it worked in my unittest

Reply:
> Can you pls create test for the case, 

I did add a test for the case, it's the last file in the PR:
https://github.com/dmitry-bym/JsonKnownTypes/blob/4e99657912ef8ef8c17ec7bd8573a9b64ab2a390/tests/JsonKnownTypes.UnitTests/WithNameDiscriminatorResolverTests.cs

Reply:
@RonSijm nope its not null cause there is default value there
```c#
public static class JsonKnownTypesSettingsManager<T>
    {
        /// <summary>
        /// Type specific default settings for discriminator
        /// </summary>
        // ReSharper disable once StaticMemberInGenericType - Justification: Intentional property should exist on a per-type base
        public static JsonDiscriminatorSettings DefaultDiscriminatorSettings { get; } = new();
    }
```